### ISSUE TITLE
Graph Editor: missing validation messages / information

### ISSUE THREAD

**Issue by marwie** [AcademySoftwareFoundation/MaterialX#1451]
Validation renders reason for why a node or graph is invalid or can't be built

Actual: no exact reason is given why the nodedef can't be built
![20230816-113052_MaterialX_Graph_Editor-needle](https://github.com/AcademySoftwareFoundation/MaterialX/assets/5083203/956a8b95-6c22-4cc4-b7ae-c4a793879177)

Expected: invalid inputs are printed
![image](https://github.com/AcademySoftwareFoundation/MaterialX/assets/5083203/a148c220-d2f2-4e26-ab79-f065c2e0c663)

Example graph that does not print an error that `SurfaceDescription_Emission` is not a valid input name
```mtlx
<?xml version="1.0"?>
<materialx name="" version="1.38" >
	<nodegraph name="needle_shadergraph" >
		<input name="_MyColor" type="color4" value="0.8509804, 0.4282091, 0.1254901, 1.0" />
		<constant name="_MyColor_constant" type="color4" >
			<input name="value" type="color4" interfacename="_MyColor" />
		</constant>
		<convert name="_MyColor_constant_to_base_color" type="color3" >
			<input name="in" type="color4" nodename="_MyColor_constant" />
		</convert>
		<output name="_MyColor_constant_to_base_color_out" type="color3" nodename="_MyColor_constant_to_base_color" />
		<input name="_MyColor" type="color4" value="0.8509804, 0.4282091, 0.1254901, 1.0" />
		<constant name="_MyColor_constant" type="color4" >
			<input name="value" type="color4" interfacename="_MyColor" />
		</constant>
	</nodegraph>
	<standard_surface name="needle_standard_surface" type="surfaceshader" >
		<input name="base_color" type="color3" nodegraph="needle_shadergraph" output="_MyColor_constant_to_base_color_out" />
		<input name="specular_roughness" type="float" value="0.0" />
		<input name="SurfaceDescription_Emission" type="vector3" value="0.7924528, 0.1607334, 0.1607334" />
		<input name="metalness" type="float" value="1.0" />
	</standard_surface>
	<surfacematerial name="Default" type="material" >
		<input name="surfaceshader" type="surfaceshader" nodename="needle_standard_surface" />
	</surfacematerial>
</materialx>
```

**Comment by jstone-lucasfilm** [comment]
Thanks for the detailed report, @marwie, and I've proposed a fix for this issue in #1468.

**Comment by jstone-lucasfilm** [comment]
The fix is now merged to main, and let us know how it works out in your local build.
